client/app/components/ApplicationArea/routeWithApiKeySession.jsx

```diff
@@ -38,9 +39,9 @@ function ApiKeySessionWrapper({ apiKey, currentRoute, renderChildren }) {
   }
 
   return (
-    <React.Fragment key={currentRoute.key}>
+    <DialogHost key={currentRoute.key}>
```

Comment:
Is there any difference in your mind per route type (api/user session) to have separate `DialogHost`'s here instead of just the `Default` one (or just have it in the ApplicationArea as it is used by all routes)?

Reply:
Is it to get any Context the `Router` may provide?

Reply:
The idea of Default host is to just keep existing dialogs working as before. If one day we'll provide a host explicitly to all `.showModal` calls - we could just replace it with a regular dialog host and remove Default host component at all. 

I added nested dialog hosts in session wrappers for several purposes: 

1. allow access to router context (as you mentioned);
2. to have an example of how it could be used.

E.g. in internal fork we have one more page wrapper with its own context, and dialogs will have to use it. Dialog host itself is a "cheap" component, so I see no problem if we'll have 1-2 more of them.